conformance/lang/expressions/literals/int-literal.balt

```diff
@@ -0,0 +1,470 @@
+Test-Case: output
+Description: Test DecimalNumbers with int-type-descriptor.
+Labels: int
+
+public function main() {
+    int x = 101;
+    io:println(x); // @output 101
+
+    x = 0;
+    io:println(x); // @output 0
+
+    x = 3;
+    io:println(x); // @output 3
+
+    x = 9223372036854775807;
+    io:println(x); // @output 9223372036854775807
+}
+
+Test-Case: output
+Description: Test DecimalNumbers with function call.
+Labels: int
+
+public function main() {
+    printInt(101);
+
+    printInt(0);
+
+    printInt(3);
+
+    printInt(9223372036854775807);
+}
+
+function printInt(int x) {
+    io:println(x); // @output 101
+                   // @output 0
+                   // @output 3
+                   // @output 9223372036854775807
+}
+
+Test-Case: error
+Description: Test DecimalNumbers with leading zeros.
+Labels: int
+
+public function main() {
+    int x = 00; // @error invalid DecimalNumber with leading '0'
+
+    x = 012345; // @error invalid DecimalNumber with leading '0'
+}
+
+Test-Case: output
+Description: Test HexIntLiterals with int-type-descriptor and function call.
+Labels: int
+
+public function main() {
+    int x = 0x1;
+    printInt(x);
+
+    x = 0xa;
+    printInt(x);
+
+    x = 0xA;
+    printInt(x);
+
+    x = 0x000000a;
+    printInt(x);
+
+    x = 0x1fed12cba7;
+    printInt(x);
+
+    x = 0xABCDEF81;
+    printInt(x);
+
+    x = 0x192AAce;
+    printInt(x);
+
+    x = 0X1;
+    printInt(x);
+
+    x = 0Xa;
+    printInt(x);
+
+    x = 0XA;
+    printInt(x);
+
+    x = 0XABCDEF81;
+    printInt(x);
+
+    printInt(0X1fedcba7);
+
+    printInt(0X192AAc5);
+}
+
+function printInt(int x) {
+    io:println(x); // @output 1
+                   // @output 10
+                   // @output 10
+                   // @output 10
+                   // @output 137121418151
+                   // @output 2882400129
+                   // @output 26389198
+                   // @output 1
+                   // @output 10
+                   // @output 10
+                   // @output 2882400129
+                   // @output 535677863
+                   // @output 26389189
```

Comment:
This is less readable. Should we break this into separate test cases or should we print inline? @MaryamZi WDYT?

Reply:
I think we decided during a review to move the output comments to wherever this function is called from.


e.g.,
```ballerina
public function main() {
    int a = 1;
    printInt(a); // 1

    printInt(2); // 2
}

function printInt(int x) {
    io:println(x);
}
```

Reply:
will update.